prov/efa/src/rxr/rxr_ep.c

```diff
@@ -2638,3 +2638,79 @@ void rxr_ep_record_tx_op_completed(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_
 	}
 }
 
+#if EFA_UNIT_TEST
+
+#include "efa_unit_tests.h"
+
+struct efa_resource {
+	struct fi_info *info;
+	struct fid_fabric *fabric;
+	struct fid_domain *domain;
+};
+
+static void delete_efa_resource(struct efa_resource *resource)
+{
+	fi_close(&resource->domain->fid);
+	fi_close(&resource->fabric->fid);
+	fi_freeinfo(resource->info);
+}
+
+static int create_efa_resource(struct efa_resource *resource)
+{
+	int ret;
+	ret = fi_getinfo(FI_VERSION(1, 14), NULL, NULL, 0ULL, NULL, &resource->info);
+	if (ret)
+		return ret;
+
+	ret = fi_fabric(resource->info->fabric_attr, &resource->fabric, NULL);
+	if (ret) {
+		fi_freeinfo(resource->info);
+		return ret;
+	}
+
+	ret = fi_domain(resource->fabric, resource->info, &resource->domain, NULL);
+	if (ret) {
+		fi_close(&resource->fabric->fid);
+		fi_freeinfo(resource->info);
+		return ret;
+	}
+
+	return 0;
+}
+
+static void check_ep_pkt_pool_flags(struct efa_resource resource, int expected_flags)
+{
+	struct fid_ep *ep;
+	struct rxr_ep *rxr_ep;
+	int ret;
+	ret = fi_endpoint(resource.domain, resource.info, &ep, NULL);
+	assert_int_equal(ret, 0);
+	rxr_ep = container_of(ep, struct rxr_ep, util_ep.ep_fid);
+	assert_int_equal(rxr_ep->efa_tx_pkt_pool->attr.flags, expected_flags);
+	assert_int_equal(rxr_ep->efa_rx_pkt_pool->attr.flags, expected_flags);
+	fi_close(&ep->fid);
+}
+
+/**
+ * @brief Test the pkt pool flags in rxr_ep_init()
+ * 
+ */
+void test_rxr_ep_pkt_pool_flags()
+{
+	int ret = 0;
+	struct efa_resource resource = {0};
+	ret = create_efa_resource(&resource);
+	assert_int_equal(ret, 0);
+
+	g_efa_fork_status = EFA_FORK_SUPPORT_ON;
+	check_ep_pkt_pool_flags(resource, OFI_BUFPOOL_NONSHARED);
```

Comment:
Maybe in a separate test, but we need to verify that when flag OFI_BUFPOOL_NONSHARED is set, the pointer is  aligned on page boundary.

Reply:
Is there any existing `ofi_` function to check that?

Reply:
Never mind, I will just check it in the test.